File: bindings/java/examples/BasicUsage.java
import com.satox.sdk.BlockchainManager;
import com.satox.sdk.NFTManager;
import com.satox.sdk.IPFSManager;
import com.satox.sdk.models.BlockchainConfig;
import com.satox.sdk.models.NFTMetadata;
import com.satox.sdk.models.Transaction;

public class BasicUsage {
    public static void main(String[] args) {
        try {
            // Initialize Blockchain Manager
            BlockchainConfig config = new BlockchainConfig.Builder()
                .setNetwork("mainnet")
                .setNodeUrl("https://node.satox.com")
                .build();
            
            BlockchainManager blockchainManager = new BlockchainManager(config);
            blockchainManager.connect();

            // Get blockchain information
            long blockHeight = blockchainManager.getBlockHeight();
            System.out.println("Current block height: " + blockHeight);

            // Create and broadcast a transaction
            Transaction tx = blockchainManager.createTransaction(
                "recipient_address",
                1.0,
                "optional_memo"
            );
            String txId = blockchainManager.broadcastTransaction(tx);
            System.out.println("Transaction broadcast: " + txId);

            // Initialize NFT Manager
            NFTManager nftManager = new NFTManager(config);

            // Create an NFT
            NFTMetadata metadata = new NFTMetadata.Builder()
                .setName("My First NFT")
                .setDescription("A unique digital asset")
                .setImage("ipfs://Qm...")
                .addAttribute("rarity", "legendary")
                .build();

            String tokenId = nftManager.createNFT(metadata);
            System.out.println("NFT created with token ID: " + tokenId);

            // Transfer NFT
            String transferTxId = nftManager.transferNFT(
                tokenId,
                "new_owner_address"
            );
            System.out.println("NFT transferred: " + transferTxId);

            // Initialize IPFS Manager
            IPFSManager ipfsManager = new IPFSManager(config);

            // Upload file to IPFS
            String cid = ipfsManager.add("path/to/file.jpg");
            System.out.println("File uploaded to IPFS: " + cid);

            // Pin content
            ipfsManager.pin(cid);
            System.out.println("Content pinned: " + cid);

            // Get pinned content
            byte[] content = ipfsManager.get(cid);
            System.out.println("Retrieved content size: " + content.length + " bytes");

            // Cleanup
            blockchainManager.disconnect();
            nftManager.disconnect();
            ipfsManager.disconnect();

        } catch (Exception e) {
            System.err.println("Error: " + e.getMessage());
            e.printStackTrace();
        }
    }
} 

File: bindings/csharp/examples/BasicUsage.cs
using Satox.SDK;
using Satox.SDK.Models;
using System;
using System.Threading.Tasks;

class BasicUsage
{
    static async Task Main(string[] args)
    {
        try
        {
            // Initialize Blockchain Manager
            var config = new BlockchainConfig
            {
                Network = "mainnet",
                NodeUrl = "https://node.satox.com"
            };

            using var blockchainManager = new BlockchainManager(config);
            await blockchainManager.ConnectAsync();

            // Get blockchain information
            var blockHeight = await blockchainManager.GetBlockHeightAsync();
            Console.WriteLine($"Current block height: {blockHeight}");

            // Create and broadcast a transaction
            var tx = await blockchainManager.CreateTransactionAsync(
                "recipient_address",
                1.0m,
                "optional_memo"
            );
            var txId = await blockchainManager.BroadcastTransactionAsync(tx);
            Console.WriteLine($"Transaction broadcast: {txId}");

            // Initialize NFT Manager
            using var nftManager = new NFTManager(config);

            // Create an NFT
            var metadata = new NFTMetadata
            {
                Name = "My First NFT",
                Description = "A unique digital asset",
                Image = "ipfs://Qm...",
                Attributes = new[]
                {
                    new NFTAttribute { TraitType = "rarity", Value = "legendary" }
                }
            };

            var tokenId = await nftManager.CreateNFTAsync(metadata);
            Console.WriteLine($"NFT created with token ID: {tokenId}");

            // Transfer NFT
            var transferTxId = await nftManager.TransferNFTAsync(
                tokenId,
                "new_owner_address"
            );
            Console.WriteLine($"NFT transferred: {transferTxId}");

            // Initialize IPFS Manager
            using var ipfsManager = new IPFSManager(config);

            // Upload file to IPFS
            var cid = await ipfsManager.AddAsync("path/to/file.jpg");
            Console.WriteLine($"File uploaded to IPFS: {cid}");

            // Pin content
            await ipfsManager.PinAsync(cid);
            Console.WriteLine($"Content pinned: {cid}");

            // Get pinned content
            var content = await ipfsManager.GetAsync(cid);
            Console.WriteLine($"Retrieved content size: {content.Length} bytes");

            // Get NFTs by owner
            var nfts = await nftManager.GetNFTsByOwnerAsync("owner_address");
            foreach (var nft in nfts)
            {
                Console.WriteLine($"Found NFT: {nft.Name} (ID: {nft.TokenId})");
            }

            // Search NFTs
            var searchResults = await nftManager.SearchNFTsAsync("legendary");
            foreach (var nft in searchResults)
            {
                Console.WriteLine($"Found NFT in search: {nft.Name}");
            }

            // Get NFT statistics
            var stats = await nftManager.GetNFTStatsAsync(tokenId);
            Console.WriteLine($"NFT stats - Views: {stats.Views}, Likes: {stats.Likes}");

        }
        catch (Exception ex)
        {
            Console.Error.WriteLine($"Error: {ex.Message}");
            Console.Error.WriteLine(ex.StackTrace);
        }
    }
} 